### PR TITLE
Backend profess&cities

### PR DESCRIPTION
**Pull request by BarBoley**
Created methods for Professions and Cities

### CODE REVIEW

File: iconic/UserProfile/urls.py [hunk 1]
```diff
@@ -8,4 +8,12 @@
...
     path('resume/edit/', views.UserResumeEditView.as_view()),
 
+    path('professions/edit', views.ProfessionsEdit.as_view(),
+         name='professions'),
+
+    path('cities/', views.CitiesGet.as_view(),
+         name='cities'),
+
+    path('cities/edit', views.CitiesEdit.as_view(),
+         name='cities'),
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/urls.py]: Delete "name" arguments

File: iconic/UserProfile/views.py [hunk 2]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+
+    def get(self, request):
+        profession = Professions.objects.get(id=request.data.id)
+        serializer = ProfessionsSerializer(profession)
+        return Response(serializer.data)
+
+    def get(self):
+        professions = Professions.objects.all()
+        serializer = ProfessionsSerializer(professions, many=True)
+        return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: All methods that process requests have arguments (self, request), also two methods that process the same request (GET) cant be in one class

File: iconic/UserProfile/views.py [hunk 3]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+
+    def get(self, request):
+        city = Cities.objects.get(id=request.data.id)
+        serializer = CitiesSerializer(city)
+        return Response(serializer.data)
+
+    def get(self):
+        cities = Cities.objects.all()
+        serializer = CitiesSerializer(cities, many=True)
+        return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: All methods that process requests have arguments (self, request), also two methods that process the same request (GET) cant be in one class

**Review by Hrustyaschiy — CHANGES_REQUESTED** [review]

File: iconic/UserProfile/serializers.py [hunk 4]
```diff
@@ -19,8 +19,23 @@ class Meta:
...
+
+    class Meta:
+        model = Professions
+        fields = ('profession_name',)
+
+
+class CitiesSerializer(serializers.ModelSerializer):
+    city_name = serializers.CharField()
+
+    class Meta:
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/serializers.py]: Names of fields in serializers should be same as names of fields inj models

File: iconic/UserProfile/views.py [hunk 5]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+        Cities.objects.get(id=request.data.id).delete()
+        return Response()
+
+
+class CitiesGet(APIView):
+
+    def get(self):
+        cities = Cities.objects.all()
+        serializer = CitiesSerializer(cities, many=True)
+        return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You forgot "request" argument

File: iconic/UserProfile/views.py [hunk 6]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+        Professions.objects.get(id=request.data.id).delete()
+        return Response()
+
+
+class ProfessionsGet(APIView):
+
+    def get(self):
+        professions = Professions.objects.all()
+        serializer = ProfessionsSerializer(professions, many=True)
+        return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You forgot "request" argument

File: iconic/UserProfile/views.py [hunk 7]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
         data = CustomResumeSerializer(resume).data
         return Response(data)
+
+
+class ProfessionsEdit(APIView):
+    
+    def get(self, request):
+        profession = Professions.objects.get(id=request.data.id)
+        serializer = ProfessionsSerializer(profession)
+        return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 8]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+    
+    def post(self, request):
+        serializer = ProfessionsSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        else:
+            return Response(serializer.errors)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 9]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+
+    def put(self, request):
+        profession = Professions.objects.get(id=request.data.id)
+        serializer = ProfessionsSerializer(profession, data=request.data)
+        if serializer.is_valid():
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 10]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+            serializer.save()
+            return Response(serializer.data)
+        else:
+            return Response(serializer.errors)
+
+    def delete(self, request):
+        Professions.objects.get(id=request.data.id).delete()
+        return Response()
+
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 11]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+        return Response(serializer.data)
+
+
+class CitiesEdit(APIView):
+
+    def get(self, request):
+        city = Cities.objects.get(id=request.data.id)
+        serializer = CitiesSerializer(city)
+        return Response(serializer.data)
+
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 12]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+    def get(self, request):
+        city = Cities.objects.get(id=request.data.id)
+        serializer = CitiesSerializer(city)
+        return Response(serializer.data)
+    
+    def post(self, request):
+        serializer = CitiesSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 13]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+        else:
+            return Response(serializer.errors)
+
+    def put(self, request):
+        city = Cities.objects.get(id=request.data.id)
+        serializer = CitiesSerializer(city, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

File: iconic/UserProfile/views.py [hunk 14]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+        else:
+            return Response(serializer.errors)
+
+    def delete(self, request):
+        Cities.objects.get(id=request.data.id).delete()
+        return Response()
+
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: You cant use data.id, use data['id'] istead of it

**Review by Hrustyaschiy — CHANGES_REQUESTED** [review]
Also i want that request returns IDs of entries

File: iconic/UserProfile/urls.py [hunk 15]
```diff
@@ -8,4 +8,9 @@
     path('resume/', views.UserResumeGetView.as_view()),
     path('resume/edit/', views.UserResumeEditView.as_view()),
 
+    path('professions/', views.ProfessionsGet.as_view()),
+    path('professions/edit', views.ProfessionsEdit.as_view()),
+
+    path('cities/', views.CitiesGet.as_view()),
+    path('cities/edit', views.CitiesEdit.as_view()),
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/urls.py]: add a slash at the end of edit in both paths

File: iconic/UserProfile/views.py [hunk 16]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
         resume = Resume.objects.get(user_id=request.user.id)
         data = CustomResumeSerializer(resume).data
         return Response(data)
+
+
+class ProfessionsEdit(APIView):
+    
+    def get(self, request):
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Want comments

File: iconic/UserProfile/serializers.py [hunk 17]
```diff
@@ -19,8 +19,23 @@ class Meta:
...
     class Meta:
         model = Resume
         fields = ('user_id', 'about_me', 'hobby', 'work_experience',
                   'achievements', 'skills', 'city', 'profession_id')
+
+
+class ProfessionsSerializer(serializers.ModelSerializer):
+    name = serializers.CharField()
+
+    class Meta:
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/serializers.py]: Want comments

File: iconic/UserProfile/views.py [hunk 18]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+
+    def get(self, request):
+        professions = Professions.objects.all()
+        serializer = ProfessionsSerializer(professions, many=True)
+        return Response(serializer.data)
+
+
+class CitiesEdit(APIView):
+
+    def get(self, request):
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Comments

File: iconic/UserProfile/views.py [hunk 19]
```diff
@@ -56,3 +56,77 @@ def get(self, request):
...
+            return Response(serializer.errors)
+
+    def delete(self, request):
+        Cities.objects.get(id=request.data['id']).delete()
+        return Response()
+
+
+class CitiesGet(APIView):
+
+    def get(self, request):
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Comments

**Review by Hrustyaschiy — CHANGES_REQUESTED** [review]

**Review by Hrustyaschiy — APPROVED** [review]

File: iconic/UserProfile/views.py [hunk 20]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
         return Response(data)
+
+#View for changing and adding a profession
+class ProfessionsEdit(APIView):
+    
+    # Method for getting profession by id
+    def get(self, request):
+        profession = Professions.objects.get(id=request.data['id'])
+        serializer = ProfessionsSerializer(profession)
+        return Response(serializer.data)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response. Look at the code from the updated dev as an example

File: iconic/UserProfile/views.py [hunk 21]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+    
+    # Method for adding profession
+    def post(self, request):
+        serializer = ProfessionsSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        else:
+            return Response(serializer.errors)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 22]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+
+ # Method for changing profession by id
+    def put(self, request):
+        profession = Professions.objects.get(id=request.data['id'])
+        serializer = ProfessionsSerializer(profession, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        else:
+            return Response(serializer.errors)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 23]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+
+ # Method for deliting profession by id 
+    def delete(self, request):
+        Professions.objects.get(id=request.data['id']).delete()
+        return Response()
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 24]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+
+
+#View for getting all professions
+class ProfessionsGet(APIView):
+# Method for getting all professions
+    def get(self, request):
+        professions = Professions.objects.all()
+        serializer = ProfessionsSerializer(professions, many=True)
+        return Response(serializer.data)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 25]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+    
+#View for changing and adding a city
+class CitiesEdit(APIView):
+
+    
+# Method for getting city by id
+    def get(self, request):
+        city = Cities.objects.get(id=request.data['id'])
+        serializer = CitiesSerializer(city)
+        return Response(serializer.data)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 26]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+    
+# Method for adding city by id
+    def post(self, request):
+        serializer = CitiesSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py:118]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 27]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+        else:
+            return Response(serializer.errors)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 28]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+
+ # Method for changing city by id 
+    def put(self, request):
+        city = Cities.objects.get(id=request.data['id'])
+        serializer = CitiesSerializer(city, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        else:
+            return Response(serializer.errors)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 29]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+
+ # Method for deliting city by id  
+    def delete(self, request):
+        Cities.objects.get(id=request.data['id']).delete()
+        return Response()
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

File: iconic/UserProfile/views.py [hunk 30]
```diff
@@ -56,3 +56,90 @@ def get(self, request):
...
+
+
+#View for getting all cities
+class CitiesGet(APIView):
+    
+#Method for getting all cities
+    def get(self, request):
+        cities = Cities.objects.all()
+        serializer = CitiesSerializer(cities, many=True)
+        return Response(serializer.data)
```
Comment (Hordd-Scratch) [inline iconic/UserProfile/views.py]: I want the status of the response.

**Review by Hordd-Scratch — APPROVED** [review]

**Review by Hrustyaschiy — APPROVED** [review]

**Pull request merged**